plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/SQLDialectQueryGenerator.java

```diff
@@ -88,6 +89,31 @@ void appendConditionString(@NotNull DBDDataFilter filter,
         boolean inlineCriteria,
         boolean subQuery);
 
+    /**
+     * Applying filters changes query formatting (thus it changes column names in expressions)
+     * Solution - always wrap query in subselect + add patched WHERE and ORDER
+     * It is configurable
+     * @param monitor database progress monitor
+     * @param dataSource datasource
+     * @param sqlQuery user query
+     * @param dataFilter filter conditions
+     * @return modified query
+     */
+    String addFiltersToQuery(
```

Comment:
This method's name is confusing. It says that it `add`s something to something, but it returns `String`.